### PR TITLE
Added score calculator on front

### PR DESCRIPTION
**Pull request by Cheshiriks**
Снимок
Снимок2

### CODE REVIEW

**Comment by nulls** [comment]
can also add calculated value for level: Low\Medium\High\Critical

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/cvsscalculator/VectorTypes.kt [hunk 1]
```diff
@@ -77,5 +82,19 @@ enum class CiaType(val value: String) {
...
+}
+
+/**
+ * Version of CVSS
+ *
+ * @property value abbreviated value from the cvss version
+ */
+@Serializable
+enum class CvssVersion(val value: String) {
+    FOUR("4.0"),
```
Comment (nulls) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/cvsscalculator/VectorTypes.kt:96]: it's not released yet, suggestion to no support now

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt [hunk 2]
```diff
@@ -275,43 +298,51 @@ val createVulnerabilityView: VFC = VFC {
...
+
                                         input {
-                                            type = InputType.number
-                                            className = ClassName("form-control")
+                                            type = InputType.text
                                             required = true
-                                            defaultValue = 0
+                                            value = vulnerability.progress
+                                            disabled = true
+                                            className = ClassName("col-2 form-control")
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt:321]: text-center

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt [hunk 3]
```diff
@@ -275,43 +298,51 @@ val createVulnerabilityView: VFC = VFC {
...
+                                        className = ClassName("row px-2")
+
                                         input {
-                                            type = InputType.number
-                                            className = ClassName("form-control")
+                                            type = InputType.text
                                             required = true
-                                            defaultValue = 0
+                                            value = vulnerability.progress
+                                            disabled = true
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt:320]: style = jso {
    backgroundColor = "rdjgruj".unsafeCast<Blabla> 
}

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        }
+        +name
+    }
+}
+
+private fun getCriticality(value: Float): String = when (value) {
+    in 0f..3.9f -> "Low"
+    in 3.9f..6.9f -> "Medium"
+    in 6.9f..8.9f -> "High"
+    in 8.9f..10f -> "Critical"
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt:333]: https://ru.wikipedia.org/wiki/Common_Vulnerability_Scoring_System
I guess should be 
```suggestion
private fun getCriticality(value: Float): String = when (value) {
    0f -> "N/A"
    in 0.1f..3.9f -> "Low"
    in 4.0f..6.9f -> "Medium"
    in 7.0f..8.9f -> "High"
    in 9.0f..10f -> "Critical"
    else -> throw IllegalStateException("Progress should be in [0; 10.0], got $value")
}
```

**Review by akuleshov7 — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/cvsscalculator/BaseMetrics.kt [hunk 5]
```diff
@@ -24,4 +24,44 @@ data class BaseMetrics(
...
+     * @return severity score vector
+     */
+    fun scoreVectorString() =
+            "$CVSS_VERSION:${version.value}/$ATTACK_VECTOR:${attackVector.value}/" +
+                    "$ATTACK_COMPLEXITY:${attackComplexity.value}/$PRIVILEGES_REQUIRED:${privilegeRequired.value}/" +
+                    "$USER_INTERACTION:${userInteraction.value}/$SCOPE:${scopeMetric.value}/" +
+                    "$CONFIDENTIALITY:${confidentiality.value}/$INTEGRITY:${integrity.value}/$AVAILABILITY:${availability.value}"
+
+    companion object {
+        const val ATTACK_COMPLEXITY = "AC"
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/cvsscalculator/BaseMetrics.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt:334]: Changes

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/cvsscalculator/VectorTypes.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/cvsscalculator/VectorTypes.kt]: Removed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt:325]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt]: Fixed

**Pull request merged**